packages/skygear-core/src/url.ts

```diff
@@ -43,11 +43,13 @@ export function decodeQueryComponent(s: string): string {
 /**
  * @public
  */
-export function decodeQuery(query?: string): [string, string][] {
-  if (!query) {
+export function decodeQuery(path?: string): [string, string][] {
```

Comment:
Making `decodeQuery` to support fragment would complicate the function. I suggest we ensure the caller only pass valid query to `decodeQuery`. Currently the only caller just split the URL by `?` so we have this bug. Maybe we can introduce a new function to split URL into 3 parts

```typescript
// splitQueryFragment splits url into 3 parts, returned as a 3-tuple.
// If query or fragment is absent, the corresponding part is an empty string.
function splitQueryFragment(url: string): [string, string, string] {
  // ...
}
```

Reply:
I remembered we discussed on polyfill for `URL` last time: https://github.com/SkygearIO/skygear-SDK-JS/pull/532#discussion_r339457614

I just searched a bit more and seems core-js has a pure version of URL polyfill: `core-js-pure/features/url`. It doesn't pollute global namespace, so we may look into using it.

Reply:
So we may add `core-js-pure` as `skygear-core` dependency, looks good to me

Reply:
Note that we want to bundle the polyfill so it should be a dev dependency and the consumer of the SDK does not need to install it. We also need to verify we only bundle the URL polyfill and nothing else.